ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueSubmission.scala

```diff
@@ -44,14 +49,26 @@ class KeyValueSubmission(metrics: Metrics) {
       tx: SubmittedTransaction,
   ): DamlSubmission =
     metrics.daml.kvutils.submission.conversion.transactionToSubmission.time { () =>
-      val inputDamlStateFromTx = InputsAndEffects.computeInputs(tx, meta)
+      val packageDamlStates = meta.optUsedPackages
+        .getOrElse(
+          throw new InternalError("Transaction was not annotated with used packages")
+        )
+        .map(Conversions.packageStateKey)
+      val inputDamlStatesFromTx = new util.ArrayList[DamlStateKey]()
+      transaction.Inputs.collectInputs(
+        tx,
+        { cid => inputDamlStatesFromTx.add(Conversions.contractIdToStateKey(cid)); () },
```

Comment:
Why not use native Scala types here for improving readability and calling `.asJava` when constructing the protobuf message?

Reply:
I apply the suggestion in  #9429